### ISSUE TITLE
How to debug with vs2022

### ISSUE THREAD

**Issue by wjkhappy14** [dapr/dotnet-sdk#793]
when  run the:dapr run --app-id routing --app-port 5000 -- dotnet run    successfully

then how to debug the  app  with vs2022 or vscode ?

**Comment by modabas** [comment]
https://github.com/dapr/dotnet-sdk/issues/401#issuecomment-747563695 worked with vs 2019 for me. I used second method (VS Extension)

**Comment by richbeales** [comment]
MSFT is there any plans to integrate this more 'obviously' into VS 2022?  Rather than 3rd party extensions.

**Comment by halspang** [comment]
@wjkhappy14 - You can also start dapr separately from your app as it will wait for your app port to start listening to finish starting. You can then debug your app normally.

@richbeales - Not that I'm aware of, but we can take it into consideration. What kind of features/abilities would you want from a Dapr extension in VS?

**Comment by DaleyKD** [comment]
I'm with @richbeales here. I'm just starting to investigate how I can use Dapr with my current microservices, which is quite a few different .sln's in Visual Studio 2022. All the steps around Dapr, having to use manual CLI all the time, etc., sure makes me not interested in using it.

I realize, as developers, we're going backwards (used to be DOS/CLI, then we had nice GUI, but now we're back to CLI). I'm trying desperately to keep the inner loop VERY simple for my teams.

Until I spend time investigating Dapr, I have zero input in how to make this easier.

**Comment by richbeales** [comment]
Yes @DaleyKD we've become accustomed to Visual Studio giving us "Right click to add feature x" (Dapr in this case), then "F5 to run" the project.

VS's key advantage over the terminal/VSCode is it should be able to bootstrap this stuff for you (and keep that tight debugging loop)

**Comment by WhitWaldo** [comment]
I'm going to go ahead and close this issue as I don't work for Microsoft and can't impact how VS supports Dapr debugging.

However, I'll leave you with a few notes:
- All packages publish symbols with their NuGet packages so you should be able to just step through the SDKs themselves, but this will prove a problem once you hit the Daprd boundary.
- Aspire's built-in tracing model makes it easier to see the requests going to and from Dapr when you use it to load your projects/Dapr
- With regards to writing better tests, the .NET SDK in 1.17 now ships with [Testcontainers support](https://v1-17.docs.dapr.io/developing-applications/sdks/dotnet/dotnet-guidance/dotnet-guidance-testcontainers/) via the [Dapr.Testcontainers](https://www.nuget.org/packages/Dapr.Testcontainers) package on NuGet to make it easier to load up your test method and Dapr in a container and perform more elaborate integration testing

I hope this at least helps in developing your applications atop Dapr. If you've got any suggestions for improving your debugging experience, I'd certainly welcome them. Please don't hesitate to open a new issue.
